src/sysfiles.c

```diff
@@ -2118,8 +2118,13 @@ int GAP_rl_func(int count, int key)
          /* display matches */
          if (!IS_LIST(data)) return 0;
          dlen = LEN_LIST(data);
-         char **strs = (char**)malloc((dlen+1) * sizeof(char*));
+         // +2, so we have a null at the end, as this must be
+         // in 'argv' format
+         char **strs = (char**)calloc(dlen+2, sizeof(char*));
          max = 0;
+         // TODO: Make this the partially typed word
```

Comment:
that's rather cryptic. Perhaps a tiny bit more context could be given? E.g. perhaps mentioning `rl_display_match_list` would be helpful?

Reply:
done. I've also made clear that, to be honest, I'm not 100% sure. I deduced what there should be by reading the source of rl_display_match_list in readline 7, as this function seems to have no (useful) documentation I can find.